code/controllers/subsystem/jobs.dm

```diff
@@ -574,36 +574,56 @@ SUBSYSTEM_DEF(jobs)
 		SSblackbox.record_feedback("nested tally", "job_preferences", young, list("[job.title]", "young"))
 		SSblackbox.record_feedback("nested tally", "job_preferences", disabled, list("[job.title]", "disabled"))
 
-
+//fuck
 /datum/controller/subsystem/jobs/proc/CreateMoneyAccount(mob/living/H, rank, datum/job/job)
-	var/datum/money_account/M = create_account(H.real_name, rand(50,500)*10, null)
-	var/remembered_info = ""
+	var/datum/money_account/account
+	if(job?.department_account_access)
+		account = GLOB.station_money_database.create_account(H.real_name, COMMAND_MEMBER_STARTING_BALANCE, ACCOUNT_SECURITY_ID, "NAS Trurl Accounting", TRUE)
+	else
+		account = GLOB.station_money_database.create_account(H.real_name, CREW_MEMBER_STARTING_BALANCE, ACCOUNT_SECURITY_ID, "NAS Trurl Accounting", TRUE)
+
+	for(var/datum/job_objective/objective as anything in H.mind.job_objectives)
+		objective.owner_account = account
 
-	remembered_info += "<b>Your account number is:</b> #[M.account_number]<br>"
-	remembered_info += "<b>Your account pin is:</b> [M.remote_access_pin]<br>"
-	remembered_info += "<b>Your account funds are:</b> $[M.money]<br>"
+	var/remembered_info = ""
+	remembered_info += "<b>Your account number is:</b> #[account.account_number]<br>"
+	remembered_info += "<b>Your account pin is:</b> [account.account_pin]<br>"
 
-	if(M.transaction_log.len)
-		var/datum/transaction/T = M.transaction_log[1]
-		remembered_info += "<b>Your account was created:</b> [T.time], [T.date] at [T.source_terminal]<br>"
 	H.mind.store_memory(remembered_info)
+	H.mind.initial_account = account
+
+	//add them to their department datum, (this relates a lot to money account I promise)
+	var/list/users_departments = get_departments_from_job(job.title)
+	for(var/datum/station_department/department as anything in users_departments)
+		var/datum/department_member/member = new()
+		member.name = H.real_name
+		member.role = job.title
+		member.member_account = account
+		member.can_approve_crates = job?.department_account_access
+		department.members += member
+
+	spawn(0)
```

Comment:
Why is this spawn here.... it was already here but I see no reason for it to be here

Reply:
im not sure, I removed it